nats-jetstream-http-connector/main.go

```diff
@@ -15,52 +15,96 @@ import (
 	"github.com/fission/keda-connectors/common"
 )
 
-var (
-	FISSION_CONSUMER = "fission_consumer" // TODO: need to change this
-)
-
 type jetstreamConnector struct {
-	host          string
-	connectordata common.ConnectorMetadata
-	jsContext     nats.JetStreamContext
-	logger        *zap.Logger
+	host            string
+	fissionConsumer string
+	connectordata   common.ConnectorMetadata
+	jsContext       nats.JetStreamContext
+	logger          *zap.Logger
+	consumer        string
 }
 
-func (conn jetstreamConnector) consumeMessage() {
+func main() {
 
-	sub, err := conn.jsContext.PullSubscribe(os.Getenv("TOPIC"), FISSION_CONSUMER, nats.PullMaxWaiting(512))
+	host := os.Getenv("NATS_SERVER")
+	consumer := os.Getenv("CONSUMER")
+
+	logger, err := zap.NewProduction()
 	if err != nil {
-		conn.logger.Fatal("error occurred while consuming message", zap.Error(err))
-	}
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-
-	signalChan := make(chan os.Signal, 1)
-	signal.Notify(signalChan, os.Interrupt)
-
-	for {
-		select {
-		case <-signalChan:
-			ctx.Done()
-			conn.logger.Info("Received an interrupt, unsubscribing and closing connection...")
-			err = sub.Unsubscribe()
-			if err != nil {
-				conn.logger.Error("error occurred while unsubscribing", zap.Error(err))
-			}
-			err = conn.jsContext.DeleteConsumer(os.Getenv("STREAM"), FISSION_CONSUMER)
-			if err != nil {
-				conn.logger.Error("error occurred while closing connection", zap.Error(err))
-			}
-			return
-		default:
-		}
-		msgs, _ := sub.Fetch(10, nats.Context(ctx))
-		for _, msg := range msgs {
-			conn.handleHTTPRequest(msg)
+		log.Fatalf("can't initialize zap logger: %v", err)
+	}
 
-		}
+	nc, _ := nats.Connect(host)
+	js, err := nc.JetStream()
+	if err != nil {
+		logger.Fatal("error while getting jetstream context:", zap.Error(err))
+	}
+
+	defer func() {
+		logger.Sync()
+	}()
+
+	connectordata, err := common.ParseConnectorMetadata()
+	if err != nil {
+		logger.Fatal("error occurred while parsing metadata", zap.Error(err))
 	}
 
+	err = validateStream(js, connectordata.Topic, logger)
+	if err != nil {
+		logger.Fatal("error occurred while validating streams", zap.Error(err))
+	}
+	// input stream
+	err = validateStream(js, connectordata.ResponseTopic, logger)
```

Comment:
Should we skip validation for ResponseTopic ? What if string is empty ?

Reply:
This would be required in usecase where we want to trigger function via nats message but dont want to capture response in either response/error topic.

Reply:
If responseTopic/errorTopic not mentioned connector will fail.

Reply:
we can skip this validation all together. Because Jetstream itself handles it and provide proper error message which we are already capturing. And taking accurate action-
- In case of input stream we break the program and return
- In case of response and error we just log it and move forward.